From 5a31e1fdc8db008fa466c499cca3ed9564654e05 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 18 Jan 2016 17:37:41 +0100
Subject: [PATCH] Fix undefined behavior

It is unlikely that ck-log-system-start would crash because of this, but
still it is better to have it fixed.
---
 src/ck-log-event.h          | 2 +-
 tools/ck-log-system-start.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ck-log-event.h b/src/ck-log-event.h
index 65571f0..4e8b29f 100644
--- a/src/ck-log-event.h
+++ b/src/ck-log-event.h
@@ -128,7 +128,7 @@ typedef struct
 {
         union {
                 CkLogNoneEvent none;
-                CkLogSystemRestartEvent system_start;
+                CkLogSystemStartEvent system_start;
                 CkLogSystemStopEvent system_stop;
                 CkLogSystemRestartEvent system_restart;
                 CkLogSeatAddedEvent seat_added;
diff --git a/tools/ck-log-system-start.c b/tools/ck-log-system-start.c
index 0860429..9726a4d 100644
--- a/tools/ck-log-system-start.c
+++ b/tools/ck-log-system-start.c
@@ -192,7 +192,7 @@ main (int    argc,
 
         event.type = CK_LOG_EVENT_SYSTEM_START;
         g_get_current_time (&event.timestamp);
-        e = (CkLogSystemStartEvent *) &event;
+        e = (CkLogSystemStartEvent *) &event.event;
 
         if (uname (&uts) == 0) {
                 e->kernel_release = uts.release;
-- 
2.5.0

